### PR TITLE
Add cane requirement and negative speed modifier for bad back quirk

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

The bad back quirk will now impose a negative speed modifier while it is active. (plus the negative mood modifier)  These can be bypassed if someone is holding a cane like item or if they choose to not wear a backpack.  To clear up confusion:

- Wearing a backpack **WITHOUT** a cane like item results in a negative mood & speed modifier
- Wearing a backpack **WITH** a cane like item results in no effect
- Wearing no backpack results in no effect

Types of cane like item include:  staves, scythes, brooms, mops, spears, pitchforks

## Why It's Good For The Game

Having a cane requirement for bad backs is thematically appropriate.

## Changelog

:cl:
balance: Bad back quirk now has a negative speed modifier while it's active and requires the person to be holding a cane like item to negate the effects. (or if they choose to not wear a backpack)  Types of canes include: staves, scythes, brooms, mops, spears, pitchforks
/:cl:

### CODE REVIEW

**Comment by newfren** [comment]
so is this only while the bag is on their back?

**Comment by timothymtorres** [comment]
> so is this only while the bag is on their back?

Yes.  Since I added an extra method to bypass the quirk (holding a cane) I made the negative effects more severe (speed modifier).  If you don't have a bag, then you are unaffected.

**Comment by calliophis-maculiceps** [comment]
great change but does the cane also remove the mood debuff? because it totally should

**Comment by JohnFulpWillard** [comment]
Odd that only 2 canes work, why is this better than what blindness does, they don't spawn with the item, but any works?

**Comment by Iamgoofball** [comment]
i'm already penalized by not being able to wear a backpack why do i have to have a cane now too? make this it's own quirk

**Comment by newfren** [comment]
you don't have to have a cane if you aren't wearing a backpack, if I understand this right

**Comment by Iamgoofball** [comment]
yeah tim explained it in discord it makes more sense now 👍

**Comment by timothymtorres** [comment]
> Odd that only 2 canes work, why is this better than what blindness does, they don't spawn with the item, but any works?

Uhhh no.  My code checks if they have a cane in either hand.  This quirk is more like paraplegic that spawns with a wheelchair.  Having to hold the cane is a hinderance, but you can always ignore it or get rid of the backpack.  It gives players flexibility in how and when they want to mitigate the effects.

I also updated the OP to clear up the confusion on how the penalties apply.  I hope this answers everyone's questions.

**Comment by anconfuzedrock** [comment]
Comically missing the point. Bad back isn't for "hmhmurff im old man" it was rock trying to think of a reason for people to not use a backpack, because the straps on tg are disgusting just remove them already and there are other back options that never get seen.

**Comment by timothymtorres** [comment]
Code is ready for review now.  I also added a whitelist of objects that can be used as a cane which should result in some interesting situations.

File: code/datums/quirks/negative.dm [hunk 1]
```diff
@@ -11,44 +11,188 @@
...
+		/obj/item/gun/magic/staff,
+		/obj/item/nullrod/staff,
+		/obj/item/nullrod/scythe,
+		/obj/item/nullrod/claymore/bostaff,
+		/obj/item/nullrod/pitchfork,
+		/obj/item/nullrod/egyptian,
+		/obj/item/nullrod/spear,
+		/obj/item/spear, // why yes, even the common spear is valid
+		/obj/item/pushbroom,
+		/obj/item/mop,
```
Comment (Time-Green) [inline code/datums/quirks/negative.dm:32]: Can you turn this type-list into a trait you put on the object? Something like TRAIT_CAINLIKE?

Comment (timothymtorres) [reply code/datums/quirks/negative.dm:32]: That's a great idea.  Initially I was attempting to make it as an element but it looked ugly.  I forgot you could attach traits to items.

Comment (timothymtorres) [reply code/datums/quirks/negative.dm]: Applied with https://github.com/tgstation/tgstation/pull/70346/commits/aaf5e8f072631a4f55376ec625aaef2536ef2495

**Pull request closed without merging**